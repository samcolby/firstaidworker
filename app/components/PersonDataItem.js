import React, { memo } from "react";
import PropTypes from "prop-types";

import { StyleSheet } from "react-native";
import { ListItem } from "react-native-elements";

import { COLORS } from "../Constants";
import PersonDataItemLeftElement from "./PersonDataItemLeftElement";

function PeopleDataItem(props) {
  const { onPress, value } = props;

  if (onPress && typeof onPress === "function") {
    return (
      <ListItem
        leftElement={<PersonDataItemLeftElement {...props} />}
        title={value}
        titleStyle={styles.titleOnPress}
        titleProps={{ onPress }}
      />
    );
  } else {
    return (
      <ListItem
        leftElement={<PersonDataItemLeftElement {...props} />}
        title={value}
      />
    );
  }
}

PeopleDataItem.propTypes = {
  icon: PropTypes.string.isRequired,
  iconType: PropTypes.string.isRequired,
  onPress: PropTypes.func,
  title: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired
};

const styles = StyleSheet.create({
  titleOnPress: {
    color: COLORS.NAV_HINTS,
    textDecorationLine: "underline",
    opacity: 0.9
  }
});

export default memo(PeopleDataItem);
