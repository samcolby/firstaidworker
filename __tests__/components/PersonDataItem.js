import React from "react";

import { shallow } from "enzyme";

import PersonDataItem from "../../app/components/PersonDataItem";

const ICON = "ios-business";
const ICON_TYPE = "ionicons";
const TITLE = "This is the title";
const VALUE = "This is the value";

describe("PersonDataItem Component", () => {
  let component;

  beforeEach(() => {
    component = shallow(
      <PersonDataItem
        icon={ICON}
        iconType={ICON_TYPE}
        title={TITLE}
        value={VALUE}
      />
    );
  });

  it("should render without issues", () => {
    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it("displays the value", () => {
    expect(component.prop("title")).toEqual(VALUE);
  });

  it("displays the leftElement in a prop", () => {
    const leftElementComponent = component.wrap(component.prop("leftElement"));
    expect(leftElementComponent.length).toBe(1);
  });

  it("displays the icon", () => {
    const leftElementComponent = component.wrap(component.prop("leftElement"));
    expect(leftElementComponent.prop("icon")).toBe(ICON);
  });

  it("displays the iconType", () => {
    const leftElementComponent = component.wrap(component.prop("leftElement"));
    expect(leftElementComponent.prop("iconType")).toBe(ICON_TYPE);
  });

  it("displays the title", () => {
    const leftElementComponent = component.wrap(component.prop("leftElement"));
    expect(leftElementComponent.prop("title")).toBe(TITLE);
  });
});

describe("PersonDataItem Component with action", () => {
  let component;
  const onPress = jest.fn();

  beforeEach(() => {
    component = shallow(
      <PersonDataItem
        icon={ICON}
        iconType={ICON_TYPE}
        title={TITLE}
        value={VALUE}
        onPress={onPress}
      />
    );
  });

  it("should render without issues", () => {
    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it("displays the value", () => {
    expect(component.prop("title")).toEqual(VALUE);
  });

  it("displays the leftElement in a prop", () => {
    const leftElementComponent = component.wrap(component.prop("leftElement"));
    expect(leftElementComponent.length).toBe(1);
  });

  it("displays the icon", () => {
    const leftElementComponent = component.wrap(component.prop("leftElement"));
    expect(leftElementComponent.prop("icon")).toBe(ICON);
  });

  it("displays the iconType", () => {
    const leftElementComponent = component.wrap(component.prop("leftElement"));
    expect(leftElementComponent.prop("iconType")).toBe(ICON_TYPE);
  });

  it("displays the title", () => {
    const leftElementComponent = component.wrap(component.prop("leftElement"));
    expect(leftElementComponent.prop("title")).toBe(TITLE);
  });

  it("displays renders onPress as a prop", () => {
    expect(component.prop("titleProps").onPress).toBe(onPress);
  });

  it("displays renders a different style when onPress present", () => {
    expect(component.prop("titleStyle")).toBeTruthy();
  });
});
