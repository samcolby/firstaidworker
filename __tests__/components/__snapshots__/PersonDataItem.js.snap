// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonDataItem Component should render without issues 1`] = `
<Themed.ListItem
  leftElement={
    <Memo(PersonDataItemLeftElement)
      icon="ios-business"
      iconType="ionicons"
      title="This is the title"
      value="This is the value"
    />
  }
  title="This is the value"
/>
`;

exports[`PersonDataItem Component with action should render without issues 1`] = `
<Themed.ListItem
  leftElement={
    <Memo(PersonDataItemLeftElement)
      icon="ios-business"
      iconType="ionicons"
      onPress={[MockFunction]}
      title="This is the title"
      value="This is the value"
    />
  }
  title="This is the value"
  titleProps={
    Object {
      "onPress": [MockFunction],
    }
  }
  titleStyle={
    Object {
      "color": "#ff5454",
      "opacity": 0.9,
      "textDecorationLine": "underline",
    }
  }
/>
`;
